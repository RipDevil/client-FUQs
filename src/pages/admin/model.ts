import { FuqType } from 'pages/fuq/model';

export interface UserType {
  login: string;
  fuqs: FuqType[];
  password: string;
  deleted: boolean;
} // password is presented? lol?
