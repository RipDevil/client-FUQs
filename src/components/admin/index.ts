export { UsersGrip } from './grips/users';
export { FuqsGrip } from './grips/fuqs';
export { Portal } from './portal';
