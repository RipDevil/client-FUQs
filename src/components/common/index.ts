export { Credits } from './credits';
export { FuqCard } from './fuqCard';
export { Spinner } from './spinner';
