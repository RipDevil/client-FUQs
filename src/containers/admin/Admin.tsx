import * as React from 'react';
import Layout from 'containers/layout/Layout';

export interface AdminProps {}

const Admin: React.FC<AdminProps> = () => {
  return (
    <Layout>
      <b>/badmin</b>
    </Layout>
  );
};

export default Admin;
