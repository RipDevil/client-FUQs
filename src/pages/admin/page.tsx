import * as React from 'react';

import { Layout } from 'templates';
// import Spinner from 'components/common/Spinner';

const Admin: React.FC = () => {
  return (
    <>
      <Layout>
        <b>/badmin</b>
      </Layout>
    </>
  );
};

export default Admin;
