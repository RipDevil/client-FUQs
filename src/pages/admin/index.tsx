import * as React from 'react';
import { Redirect } from 'react-router-dom';
import { useStore } from 'effector-react';
import { MenuInfo } from 'rc-menu/lib/interface';
import { Menu, Col, Button, message } from 'antd';

import { AdminLayout } from 'templates';
import { UsersGrip, FuqsGrip } from 'components/admin';
import { $auth, authReset } from 'pages/login/model';
import { useLogout } from 'api';

const { Item } = Menu;

type ItemType = {
  key: number;
  title: string;
};

const DEFAULT_GRIP: string = '0';
const Admin: React.FC = () => {
  const [grip, setGrip] = React.useState<string>('Users');
  const authStore = useStore($auth);
  const [items] = React.useState<ItemType[]>([
    {
      key: 0,
      title: 'Users',
    },
    {
      key: 1,
      title: 'FUQs',
    },
  ]);

  const { mutate: logout, isError, isLoading, isSuccess } = useLogout();

  React.useEffect(() => {
    isError && message.warning('Can not log off');
  }, [isError]);

  React.useEffect(() => {
    isSuccess && authReset();
  }, [isSuccess]);

  const onItemClick = ({ key: reactKey }: MenuInfo) => {
    const key = reactKey.toString();
    setGrip(items[parseInt(key)].title);
  };

  // @mna: is this the best variant?
  if (authStore.refreshToken === '' || authStore.refreshToken === '') {
    return <Redirect to="/login" />;
  }

  return (
    <AdminLayout>
      <Col span={5}>
        <Button data-testid="logout-btn" onClick={logout} type="primary" loading={isLoading} block>
          Logout
        </Button>
        <Menu onClick={onItemClick} mode="inline" defaultSelectedKeys={[DEFAULT_GRIP]} style={{ height: '100%' }}>
          {items && items.length ? (
            items.map((item) => (
              <Item data-testid={item.key} key={item.key}>
                {item.title}
              </Item>
            ))
          ) : (
            <b>...</b>
          )}
        </Menu>
      </Col>
      <Col span={19} style={{ padding: '5px' }}>
        {grip === items[0].title && <UsersGrip />}
        {grip === items[1].title && <FuqsGrip />}
      </Col>
    </AdminLayout>
  );
};

export default Admin;
