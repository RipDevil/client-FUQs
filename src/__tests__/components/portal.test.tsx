import { render } from '@testing-library/react';
import { Router, Route } from 'react-router-dom';
import { createMemoryHistory } from 'history';

import { Portal } from 'components/admin';

beforeAll(() => {
  localStorage.setItem('fuqs-admin', 'true');
});

describe('Portal element tests', () => {
  describe('Portal should be rendered', () => {
    test('on /login or /badmin', () => {
      const { container: first } = render(
        <Router history={createMemoryHistory({ initialEntries: ['/login'] })}>
          <Route path={'/login'} component={Portal} />
        </Router>,
      );

      const { container: second } = render(
        <Router history={createMemoryHistory({ initialEntries: ['/badmin'] })}>
          <Route path={'/badmin'} component={Portal} />
        </Router>,
      );

      // TODO: isn't it too expensive?
      expect(first.innerHTML).toBe(second.innerHTML);
      expect(second).toMatchSnapshot();
    });

    test('on /else', () => {
      const { container } = render(
        <Router history={createMemoryHistory({ initialEntries: ['/else'] })}>
          <Route path={'/else'} component={Portal} />
        </Router>,
      );

      expect(container).toMatchSnapshot();
    });
  });

  it('Should not be rendered if no local storage field presented', () => {
    localStorage.removeItem('fuqs-admin');

    const { container } = render(
      <Router history={createMemoryHistory({ initialEntries: ['/badmin'] })}>
        <Route path={'/badmin'} component={Portal} />
      </Router>,
    );

    expect(container).toMatchSnapshot();
  });
});
