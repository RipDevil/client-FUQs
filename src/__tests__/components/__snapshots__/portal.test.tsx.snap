// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Portal element tests Portal should be rendered on /else 1`] = `
.c0 {
  position: fixed;
  top: 1vh;
  right: 0;
  z-index: 1001;
  overflow: hidden;
  font-size: 10px;
  -webkit-transition: 500ms ease;
  transition: 500ms ease;
}

.c0:hover {
  right: 10000 !important;
}

<div>
  <div
    class="c0"
  >
    <a
      class="ant-btn ant-btn-link ant-btn-sm"
      href="/login"
    >
      <span
        aria-label="login"
        class="anticon anticon-login"
        role="img"
        title="Login"
      >
        <svg
          aria-hidden="true"
          data-icon="login"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <defs>
            <style />
          </defs>
          <path
            d="M521.7 82c-152.5-.4-286.7 78.5-363.4 197.7-3.4 5.3.4 12.3 6.7 12.3h70.3c4.8 0 9.3-2.1 12.3-5.8 7-8.5 14.5-16.7 22.4-24.5 32.6-32.5 70.5-58.1 112.7-75.9 43.6-18.4 90-27.8 137.9-27.8 47.9 0 94.3 9.3 137.9 27.8 42.2 17.8 80.1 43.4 112.7 75.9 32.6 32.5 58.1 70.4 76 112.5C865.7 417.8 875 464.1 875 512c0 47.9-9.4 94.2-27.8 137.8-17.8 42.1-43.4 80-76 112.5s-70.5 58.1-112.7 75.9A352.8 352.8 0 01520.6 866c-47.9 0-94.3-9.4-137.9-27.8A353.84 353.84 0 01270 762.3c-7.9-7.9-15.3-16.1-22.4-24.5-3-3.7-7.6-5.8-12.3-5.8H165c-6.3 0-10.2 7-6.7 12.3C234.9 863.2 368.5 942 520.6 942c236.2 0 428-190.1 430.4-425.6C953.4 277.1 761.3 82.6 521.7 82zM395.02 624v-76h-314c-4.4 0-8-3.6-8-8v-56c0-4.4 3.6-8 8-8h314v-76c0-6.7 7.8-10.5 13-6.3l141.9 112a8 8 0 010 12.6l-141.9 112c-5.2 4.1-13 .4-13-6.3z"
          />
        </svg>
      </span>
    </a>
  </div>
</div>
`;

exports[`Portal element tests Portal should be rendered on /login or /badmin 1`] = `
.c0 {
  position: fixed;
  top: 1vh;
  right: 0;
  z-index: 1001;
  overflow: hidden;
  font-size: 10px;
  -webkit-transition: 500ms ease;
  transition: 500ms ease;
}

.c0:hover {
  right: 10000 !important;
}

<div>
  <div
    class="c0"
  >
    <a
      class="ant-btn ant-btn-link ant-btn-sm"
      href="/"
    >
      <span
        aria-label="home"
        class="anticon anticon-home"
        role="img"
        title="Main page"
      >
        <svg
          aria-hidden="true"
          data-icon="home"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M946.5 505L560.1 118.8l-25.9-25.9a31.5 31.5 0 00-44.4 0L77.5 505a63.9 63.9 0 00-18.8 46c.4 35.2 29.7 63.3 64.9 63.3h42.5V940h691.8V614.3h43.4c17.1 0 33.2-6.7 45.3-18.8a63.6 63.6 0 0018.7-45.3c0-17-6.7-33.1-18.8-45.2zM568 868H456V664h112v204zm217.9-325.7V868H632V640c0-22.1-17.9-40-40-40H432c-22.1 0-40 17.9-40 40v228H238.1V542.3h-96l370-369.7 23.1 23.1L882 542.3h-96.1z"
          />
        </svg>
      </span>
    </a>
  </div>
</div>
`;

exports[`Portal element tests Should not be rendered if no local storage field presented 1`] = `<div />`;
